dora/shell/src/main/java/alluxio/cli/fsadmin/report/UfsCommand.java

```diff
@@ -40,55 +44,45 @@ public UfsCommand(FileSystemMasterClient fileSystemMasterClient) {
    * @return 0 on success, 1 otherwise
    */
   public int run() throws IOException {
-    Map<String, MountPointInfo> mountTable = mFileSystemMasterClient.getMountTable();
-    System.out.println("Alluxio under storage system information:");
-    printMountInfo(mountTable);
-    return 0;
-  }
+    ObjectMapper mapper = new ObjectMapper();
+    ArrayNode ufsInfo = mapper.createArrayNode();
 
-  /**
-   * Prints mount information for a mount table.
-   *
-   * @param mountTable the mount table to get information from
-   */
-  public static void printMountInfo(Map<String, MountPointInfo> mountTable) {
+    Map<String, MountPointInfo> mountTable = mFileSystemMasterClient.getMountTable();
     for (Map.Entry<String, MountPointInfo> entry : mountTable.entrySet()) {
+      ObjectNode mountInfo = mapper.createObjectNode();
```

Comment:
try to represent information as classes to json serialize instead of nodes. although you have to set up the setters/getters (maybe can find some way to help generate the code), a class is more friendly to be used by different formats.
```
public class MountInfoOutput {
  public String mUri;
  public String mMountPoint;
  public String mUfsType;
  public String mTotalCapacity;
  public String mUsedCapacity;
  public boolean mReadOnly;
  public Shared mShared;

  // add setter and getter methods for json to work on this object
}
```

for the two capacity values, it would be even better to use a `long` type for the field and use an annotation to declare a custom serialization function that will execute the `FormatUtils.getSizeFromBytes`

actually i noticed that the original `MountPointInfo` object is already mostly ready to be directly used to json serialize. the only field we don't want to show is the `Properties` map. so there should be a way to add an annotation to tell the json serializer to avoid serializing the field. this is the best situation because it reuses the existing code as much as possible.

Reply:
one way to serialize the annotated field only:
1. add annotations to all the fields you need:
```java
@JsonProperty("readOnly")
public boolean mReadOnly;
```
2. set `ObjectMapper` to ignore the getters/setters and only focus on the annotation:
```java
ObjectMapper mapper = new ObjectMapper();
mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
```

Reply:
I'm not very sure about this change. Does this mean if I'd like my command to print something more, instead of a println, now I need to update the object? I mean trying to objectify printouts seems awkward unless you have a very good reason?

Reply:
Objectifying the printed output ensures consistency. Printlns are at the whims of the particular dev; the lack of consistency in general is why both the command line format (ex. Flags, arguments, command tree structure) and the output format are a total mess. Objectifying outputs also allows for various output format (ex. Json for machine parseable output, yaml for more human readable output)